module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends AbstractHelper
+{
+    /**
+     * Create an anchor tag
+     *
+     * $href will override $attrs['href']
+     * > Feel free to use like makeLink('text', 'href', $defaults);
+     *
+     * If no $href, will try to find an href in $attrs
+     * > makeLink('text', null, ['href' => '#', 'class' => 'btn-link'])
+     *
+     * If $attrs is a string, will be treated like a class
+     * > makeLink('text', $href, 'btn-link')
+     *
+     * @param string       $text  Link contents (should be properly-formed HTML)
+     * @param string|array $href  Link destination (null to skip)
+     * @param array        $attrs Link attributes (associative array)
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke($text, $href = null, $attrs = [])
```

Comment:
`$text` => `$innerHtml` or something to indicate it's HTML, not just text?

I'm ok with the current way that the content needs to be properly formed HTML, but if we wanted to play it safe (and that wouldn't hurt considering how many escaping issues I've fixed lately), I'd make it so that this would be `$content` which, if a string, would be escaped. If you wanted to use HTML, you'd have to do it explicitly e.g. by making it an array like `['html' => '<span...>']`.

Reply:
That's a pretty good idea, too -- might cut down on calls to the escape helper in some situations while, as you say, preventing accidental oversights. @crhallberg, what do you think about this one?